crates/llama-cpp-server/src/supervisor.rs

```diff
@@ -87,30 +92,64 @@ impl LlamaCppSupervisor {
                     )
                 });
 
-                let status_code = process
-                    .wait()
-                    .await
-                    .ok()
-                    .and_then(|s| s.code())
-                    .unwrap_or(-1);
+                let mut stderr =
+                    BufReader::new(process.stderr.take().expect("Failed to get stderr")).lines();
+                let mut error_lines = VecDeque::with_capacity(100);
+
+                let wait_handle = process.wait();
+
+                while let Ok(Some(line)) = stderr.next_line().await {
+                    if !line.contains("GET /health") {
+                        if error_lines.len() >= 100 {
+                            error_lines.pop_front();
+                        }
+                        error_lines.push_back(line);
+                    }
+                }
+
+                let status_code = wait_handle.await.ok().and_then(|s| s.code()).unwrap_or(-1);
 
                 if status_code != 0 {
                     warn!(
                         "llama-server <{}> exited with status code {}, args: `{}`",
                         name, status_code, command_args
                     );
-                    let mut stderr = process
-                        .stderr
-                        .take()
-                        .map(tokio::io::BufReader::new)
-                        .map(|reader| reader.lines())
-                        .expect("Failed to read stderr");
-
-                    while let Ok(Some(line)) = stderr.next_line().await {
-                        warn!("<{}>: {}", name, line);
-                    }
 
-                    tokio::time::sleep(std::time::Duration::from_secs(1)).await;
+                    let error_message = error_lines.into_iter().fold(
```

Comment:
print the error message directly later instead of accumulating it into a string?